02_cancer_type_classification/lasso_range_analysis/nbconverted/compare_optimizers_all.py

```diff
@@ -0,0 +1,318 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# ### Comparison of optimizers for LASSO parameter range experiments
+# 
+# sklearn has 2 ways to fit logistic regression models with a LASSO penalty: [LogisticRegression](https://scikit-learn.org/stable/modules/generated/sklearn.linear_model.LogisticRegression.html) (using the liblinear optimizer; i.e. coordinate descent), and [SGDClassifier](https://scikit-learn.org/stable/modules/generated/sklearn.linear_model.SGDClassifier.html) (which uses stochastic gradient descent).
+# 
+# Here, we want to compare mutation prediction results between the two optimizers, across all the cancer genes in our driver gene set.
+
+# In[1]:
+
+
+import os
+import itertools as it
+
+import numpy as np
+import pandas as pd
+import matplotlib.pyplot as plt
+from scipy.stats import pearsonr, spearmanr
+import seaborn as sns
+
+import pancancer_evaluation.config as cfg
+import pancancer_evaluation.utilities.data_utilities as du
+import pancancer_evaluation.utilities.analysis_utilities as au
+
+get_ipython().run_line_magic('load_ext', 'autoreload')
+get_ipython().run_line_magic('autoreload', '2')
+
+
+# In[2]:
+
+
+ll_base_results_dir = os.path.join(
+    cfg.repo_root, '02_cancer_type_classification', 'results', 'lasso_range_lr_all_features'
+)
+
+# this doesn't have a sex covariate but it's probably close enough
+sgd_base_results_dir = os.path.join(
+    cfg.repo_root, '02_cancer_type_classification', 'results', 'lasso_range_valid'
+)
+
+training_dataset = 'all_other_cancers'
+ll_results_dir = os.path.join(ll_base_results_dir, training_dataset)
+sgd_results_dir = os.path.join(sgd_base_results_dir, training_dataset)
+
+metric = 'aupr'
+
+# "cv" to plot performance on validation set (same cancer types as training)
+# "test" to plot performance on held-out cancer type (not in training set)
+data_type = 'cv'
+
+
+# ### Get coefficient information for each lasso penalty
```

Comment:
I'm not sure where you're using this info? My best guess is the plot at the bottom but that's sorted by "ll_sgd_diff" which I assume is the AUPR difference.

Reply:
You're right, it's not being used in this script - good catch. Loading it here for all genes takes a while, so probably a good idea to remove it unless it's really needed.